examples/slurm_cloud_deploy/deploy-gpu.yaml

```
@@ -0,0 +1,248 @@
+resources:
+  cloud: aws
+  cpus: 4+
+  disk_size: 50
+  accelerators: T4:4  # Add GPU specification - can be changed as needed (e.g., A100, T4, etc.)
+
+num_nodes: 2
+
+run: |
+  set -ex
+  sudo apt update
+  sudo apt install -y munge slurm-wlm slurm-wlm-basic-plugins slurmctld slurmd
+
+  # Skip NVIDIA driver installation as it's already installed
+  echo "Using pre-installed NVIDIA drivers"
+
+  CONTROLLER_IP=$(echo "$SKYPILOT_NODE_IPS" | head -n1)
+
+  # Define barrier function
+  create_barrier() {
+    local barrier_name=$1
+    local barrier_file="/tmp/${barrier_name}${SKYPILOT_TASK_ID}_barrier"
+    
+    if [ "$SKYPILOT_NODE_RANK" = "0" ]; then
+      # Controller node creates and distributes barrier file
+      touch $barrier_file
+      for node in $(echo "$SKYPILOT_NODE_IPS" | tail -n +2); do
+        scp -o StrictHostKeyChecking=no $barrier_file $node:$barrier_file
+      done
+    else
+      # Non-controller nodes wait for barrier file
+      while [ ! -f "$barrier_file" ]; do
+        echo "Waiting for $barrier_name barrier..."
+        sleep 5
+      done
+    fi
+  }
+
+  # Generate munge key manually on controller
+  if [ "$SKYPILOT_NODE_RANK" = "0" ]; then
+    sudo sh -c 'dd if=/dev/urandom bs=1 count=1024 > /etc/munge/munge.key'
+    sudo chown munge:munge /etc/munge/munge.key
+    sudo chmod 400 /etc/munge/munge.key
+
+    # Copy munge.key temporarily for scp
+    sudo cp /etc/munge/munge.key /tmp/munge.key
+    sudo chown ubuntu:ubuntu /tmp/munge.key
+
+    # Now you can scp from /tmp/munge.key
+    for node in $(echo "$SKYPILOT_NODE_IPS" | tail -n +2); do
+      scp -o StrictHostKeyChecking=no /tmp/munge.key ubuntu@$node:/tmp/munge.key
+      ssh -o StrictHostKeyChecking=no ubuntu@$node "sudo mv /tmp/munge.key /etc/munge/munge.key && sudo chown munge:munge /etc/munge/munge.key && sudo chmod 400 /etc/munge/munge.key"
+    done
+
+    # cleanup
+    sudo rm /tmp/munge.key
+  fi
+  
+  # Create barrier to ensure munge key is distributed before continuing
+  create_barrier "munge_key_distributed"
+
+  sudo systemctl stop munge || true
+  sudo systemctl enable munge
+  sudo systemctl start munge
```

```suggestion
  sudo systemctl enable munge
  sudo systemctl restart munge
```
should be sufficient